src/Orleans/Utils/IKeyedServiceCollection.cs

```diff
@@ -0,0 +1,37 @@
+﻿
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Collection of services that can be disambiguated by key
+    /// </summary>
+    public interface IKeyedServiceCollection<TKey, TService>
+        where TService : class
+    {
+        TService GetService(TKey key);
+    }
+
+    public static class KeyedServiceExtensions
+    {
+        /// <summary>
+        /// Acquire a service by key.
+        /// </summary>
+        public static TService GetServiceByKey<TKey, TService>(this IServiceProvider services, TKey key)
```

Comment:
Suggest TKey be last generic argument.  May be more intuitive, as user probably think about what they want to get, then what key they use to get it.

Keep generic argument order uniform across all calls.